api/src/main/java/com/community/api/services/ImageService.java

```diff
@@ -30,11 +33,54 @@ public ImageService(EntityManager entityManager) {
         this.entityManager = entityManager;
     }
 
+//    @Transactional
+//    public Image saveImage(MultipartFile file) throws Exception {
+//        // Define the directory where you want to store the images
+//        String uploadDir = "api/avisoftdocument/Random Images";
+//
+//        // Create the directory if it doesn't exist
+//        File directory = new File(uploadDir);
+//        if (!directory.exists()) {
+//            directory.mkdirs();
+//        }
+//
+//        byte[] fileBytes = file.getBytes();
+//        // Generate the file path (append the filename properly with a separator)
+//        String filePath = uploadDir + File.separator + file.getOriginalFilename();
+//
+//        try {
+//            File destFile = new File(filePath);
+//            FileUtils.writeByteArrayToFile(destFile, file.getBytes());
+//        } catch (IOException e) {
+//            throw new Exception("Failed to save the file", e);
+//        }
+//
+//        // Create and populate the Image entity
+//        Image image = new Image();
+//        image.setFile_name(file.getOriginalFilename());
+//        image.setFile_type(file.getContentType());
+//        image.setImage_data(fileBytes);
+//        image.setFile_path(filePath); // Store the file path in the database
+//
+//        // Persist the image entity to the database
+//        entityManager.persist(image);
+//        return image;
+//    }
+
     @Transactional
     public Image saveImage(MultipartFile file) throws Exception {
         // Define the directory where you want to store the images
         String uploadDir = "api/avisoftdocument/Random Images";
 
+        // Define acceptable image MIME types
+        List<String> allowedMimeTypes = Arrays.asList("image/jpeg", "image/png", "image/gif", "image/bmp", "image/tiff");
```

Comment:
isValidFileType call this function i have already create it in DocumentStorageService